### PR TITLE
Lesson15 Tests

### PR DESCRIPTION
**Pull request by olyastrilectka**

### CODE REVIEW

**Comment by CyberSergo** [comment]
Hi, Olya. Your code is fully working, I'm glad you trusted me to check it. I learned a lot of new things for myself in Cypress, but there is one problem: you have too much repetitive code. You can move some of your code into functions or classes. This will make your code easier to read and maintain.

File: cypress/integration/lesson15.spec.js [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+    cy.visit('https://demoqa.com/webtables');
+  });
+
+  describe('User creating', () => {
+    it('Should open registration form', () => {
+      // arrange
+      // act
+      cy.get('#addNewRecordButton').click();
+      // assert
+      cy.get('.modal-dialog').should('exist');
```
Comment (landriiuk) [inline cypress/integration/lesson15.spec.js]: Its good that u use AAA pattern, but delete comments

File: cypress/integration/lesson15.spec.js [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+
+describe('Lesson15', () => {
+  const firstUserCellsSelector =
+    '.rt-table .rt-tr-group:first-child .rt-tr .rt-td';
+  beforeEach(() => {
+    cy.visit('https://demoqa.com/webtables');
+  });
+
+  describe('User creating', () => {
+    it('Should open registration form', () => {
```
Comment (landriiuk) [inline cypress/integration/lesson15.spec.js]: U can use following structure:

```suggestion
describe('Lesson15', () => {
  const firstUserCellsSelector =
    '.rt-table .rt-tr-group:first-child .rt-tr .rt-td';
  beforeEach(() => {
    cy.visit('https://demoqa.com/webtables');
  });

  context('User creating', () => {
    it('Should open registration form', () => {
```

File: cypress/integration/lesson15.spec.js [hunk 3]
```diff
@@ -0,0 +1,493 @@
...
+        'rgb(220, 53, 69)',
+      );
+      cy.get('#age').should('have.css', 'border-color', 'rgb(220, 53, 69)');
+      cy.get('#salary').should('have.css', 'border-color', 'rgb(220, 53, 69)');
+      cy.get('#department').should(
+        'have.css',
+        'border-color',
+        'rgb(220, 53, 69)',
+      );
+    });
```
Comment (landriiuk) [inline cypress/integration/lesson15.spec.js:90]: missed space between the tests

File: cypress/integration/lesson15.spec.js [hunk 4]
```diff
@@ -0,0 +1,493 @@
...
+      cy.contains(department).should('exist');
+    });
+  });
+
+  describe('User editing', () => {
+    it('Should have User record edit button', () => {
+      // assert
+      cy.get('#edit-record-1').should('exist');
+    });
+    it('Should open edit modal on user edit button click', () => {
```
Comment (landriiuk) [inline cypress/integration/lesson15.spec.js:115]: and here

File: cypress/integration/lesson15.spec.js [hunk 5]
```diff
@@ -0,0 +1,493 @@
...
+      cy.get('.rt-th:nth-child(6)').dblclick();
+      cy.get('.rt-th:nth-child(6)').should('have.class', '-sort-desc');
+
+      cy.get('.rt-tr-group .rt-td:nth-child(6)')
+        .then(getInnerText)
+        .then((departments) => {
+          const manuallySortedDepartments = sortStringsDESC(departments);
+          expect(departments).to.deep.equal(manuallySortedDepartments);
+        });
+    });
```
Comment (landriiuk) [inline cypress/integration/lesson15.spec.js]: Try to create a few reusable func, and do not use a lot of copy-paste

File: utilities/data.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+const getInnerText = (cells) => cells.map((cell) => cell.innerText);
```
Comment (landriiuk) [inline utilities/data.js:1]: Do not forget about Upper Case for const

**Review by landriiuk — COMMENTED** [review]
Your code working well, pls delete your comments. 
after that you are able to merge this hw to master

Score 9/10

File: cypress/integration/lesson15.spec.js (hunk 1 above)
Comment (olyastrilectka) [reply cypress/integration/lesson15.spec.js]: thx, done

File: cypress/integration/lesson15.spec.js (hunk 2 above)
Comment (olyastrilectka) [reply cypress/integration/lesson15.spec.js]: Have changed, thanks

File: cypress/integration/lesson15.spec.js (hunk 3 above)
Comment (olyastrilectka) [reply cypress/integration/lesson15.spec.js:90]: done

File: cypress/integration/lesson15.spec.js (hunk 4 above)
Comment (olyastrilectka) [reply cypress/integration/lesson15.spec.js:115]: done

File: utilities/data.js (hunk 6 above)
Comment (olyastrilectka) [reply utilities/data.js:1]: done

**Pull request merged**